### DISCUSSION TITLE
Can you support etcd?

### DISCUSSION THREAD

**Discussion by zuochangqin** [smallrye/smallrye-stork#621] [Ideas]
Can you support etcd?

**Comment by cescoffier** [comment]
Etcd is slightly harder in the sense that it's very generic. It means that we need to define ourself how a service is registered in etcd. 

Do you know if there is some kind of standard? 

In addition, you can implement a custom service discovery for etcd using your own  opinionated way to register the services.

**Comment by aureamunoz** [comment]
Using etcd directly is not a very secure practice. Read/write data to the kubernetes etcd store is root access to every cluster node. From what I see, to use the kubernetes etcd store should be done via the kubernetes API, creating a config-map or CRD but in that case yes, as already mentioned, is there a standard?
